crates/syntax/src/ast/syntax_factory/constructors.rs

```diff
@@ -512,3 +507,13 @@ impl SyntaxFactory {
         make::tokens::whitespace(text)
     }
 }
+
+fn iterator_input<N: AstNode>(input: impl IntoIterator<Item = N>) -> (Vec<N>, Vec<SyntaxNode>) {
```

Comment:
That sounds like there should be a comment on this function then explaining the rational like the PR does